eparticlectr/eparticlectr.cpp

```diff
@@ -175,11 +54,39 @@ void eparticlectr::brainclmid( uint64_t stakeid ) {
     stake_it = staketable.erase(stake_it);
 }
 
+// Redeem IQ, with a specific stake specified
+[[eosio::action]]
+void eparticlectr::brainclmidex( uint64_t stakeid ) {
+
+    // Get the stakes
+    staketblex staketable(_self, _self.value);
+    auto stake_it = staketable.find(stakeid);
+    eosio::check( stake_it != staketable.end(), "Stake does not exist in database");
+
+    // See if the stake is complete
+    eosio::check( eosio::current_time_point().sec_since_epoch() > stake_it->completion_time, "Staking period not over yet");
+
+    // Transfer back the IQ
+    asset iqAssetPack = asset(int64_t(stake_it->amount * IQ_PRECISION_MULTIPLIER), IQSYMBOL);
+    std::string memo = std::string("return stake #") + std::to_string(stake_it->id);
+    action(
```

Comment:
I think for token balances purposes transfrextra needs to call an inline action to transfer, but brainclmidex probably could do a internal call avoiding this extra inline call.

Not big deal, but each inline action has an impact in CPU consumption

Reply:
Part of the reason I want to have it like this is so I can read transfrextra calls on chain and sync them with the EPID setup. You are probably right though that it isn't the most efficient.